### DISCUSSION TITLE
is there a way to provide a pattern for file paths to diff?

### DISCUSSION THREAD

**Discussion by andrewbanchich** [dandavison/delta#1716] [Q&A]
e.g. if i have `foo/`, `foo/abc_foo.md`, `foo/do_stuff/some_foo.txt` and also `bar/`, `bar/abc_bar.md`, `bar/do_stuff/some_bar.txt`, i'd like to be able to tell delta that the file structures are the same except for `foo` -> `bar`, so `foo/do_stuff/some_foo.txt` should be diffed with `bar/do_stuff/some_bar.txt`.

**Comment by th1000s** [comment]
`delta` works on the `git diff` output, and if git can not output that the diff was just a `foo` -> `bar` rename instead of listing all individual files then delta currently can not either. To be fast it works on a line-by-line basis without much look ahead, but renames certainly could be gathered and summarized. But this becomes complex if it is 99 renames, but also one little diff in the middle.

**Reply by andrewbanchich** [reply]
`delta` isn't just for git diffs though - it's also `diff` and `grep`.
